cmd/substreams-sink-noop/main.go

```diff
@@ -166,31 +283,45 @@ func run(cmd *cobra.Command, args []string) error {
 
 	go sinker.Run(ctx)
 
-	zlog.Info("ready, waiting for signal to quit")
-
-	signalHandler, isSignaled, _ := cli.SetupSignalHandler(0*time.Second, zlog)
 	select {
 	case <-signalHandler:
 		go app.Shutdown(nil)
-		break
 	case <-app.Terminating():
-		zlog.Info("run terminating", zap.Bool("from_signal", isSignaled.Load()), zap.Bool("with_error", app.Err() != nil))
-		break
+		zlog.Info("run terminating", zap.Bool("with_error", app.Err() != nil))
 	}
 
 	zlog.Info("waiting for run termination")
 	select {
 	case <-app.Terminated():
+		return app.Err()
 	case <-time.After(30 * time.Second):
 		zlog.Warn("application did not terminate within 30s")
+		return app.Err()
 	}
+}
+
+func computeBlockRangeFromHead(ctx context.Context, blockmetaClient pbbmsrvconnect.BlockClient, reversibleSegmentSize uint64, substreamsSegmentSize uint64, blockRangeArg string) (string, error) {
```

Comment:
I think you should get the blockHead in the run() function and keep the 'computeBlockRangeFromHead' function's purpose to actually compute it (not fetch it.)
Easier to test, clearer separation of concerns, and will allow you to use the headBlockNum in the logs I asked in my last comment.